### PR TITLE
[HttpFoundation] Same site cookie

### PR DESCRIPTION
**Pull request by cvilleger**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #25344
| License       | MIT
| Doc PR        | -

TODO : 
- [x] Symfony does not check the value of value against the characters =,; \t\r\n\013\014 whereas PHP does. Symfony only checks it against name and PHP checks it agains name and value.
- [x] Add test for invalid name (But should I add an Exception and test for invalid value ? BC ... )
- [x] When a cookie needs to be deleted, Symfony sets the max-age to 31536001 and the expiry to now
- [x] max-age but PHP just passes in expiry = 1 and max-age = 0.
- [x] PHP checks if the expiry date has a year greater than 9999.
- [x] Symfony also urlencode's name, I don't see PHP do that.
- [x] Add test for urlencode of name and value
- [x] Symfony uses urlencode for the name vs rawurlencode for the value, not sure why that difference is there.

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/response-functional/cookie_max_age.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+use Symfony\Component\HttpFoundation\Cookie;
+
+$r = require __DIR__ . '/common.inc';
+
+$r->headers->setCookie(new Cookie('foo', 'bar', 946749600, '', null, false, false));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Tests/Fixtures/response-functional/cookie_max_age.php]: 253402310800

**Comment by nicolas-grekas** [comment]
@cvilleger can you push your lastest changes please?

File: src/Symfony/Component/HttpFoundation/Tests/Fixtures/response-functional/cookie_max_age.php (hunk 1 above)
Comment (cvilleger) [reply src/Symfony/Component/HttpFoundation/Tests/Fixtures/response-functional/cookie_max_age.php]: Done, very sorry for the time taken.

**Comment by cvilleger** [comment]
When executing `./phpunit src/Symfony/Component/HttpFoundation/Tests/ResponseFunctionalTest.php `, I have one error on Max-Age attribute, trying to fix that...

```
1) Symfony\Component\HttpFoundation\Tests\ResponseFunctionalTest::testCookie with data set #0 ('cookie_max_age')
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
     [2] => Date: Sat, 12 Nov 1955 20:04:00 GMT\n
-    [3] => Set-Cookie: foo=bar; expires=Sat, 01-Jan-10000 02:46:40 GMT; Max-Age=251878181946; path=/\n
+    [3] => Set-Cookie: foo=bar; expires=Sat, 01-Jan-10000 02:46:40 GMT; Max-Age=251878181664; path=/\n
 )\n
 shutdown\n
 '

```

**Comment by nicolas-grekas** [comment]
Merged, thanks

**Pull request closed without merging**